oktapam/client/project.go

```diff
@@ -37,7 +37,7 @@ func (p Project) ToResourceMap() map[string]interface{} {
 		m[attributes.Name] = *p.Name
 	}
 	if p.ID != nil {
-		m[attributes.ID] = *p.ID
+		m[attributes.ProjectID] = *p.ID
```

Comment:
I don't think I understand this. Shouldn't this map to the resource's ID?

Reply:
Terraform Resource Id in project resource is set to "Project Name". Earlier, when I discussed with Steven the idea was to set it to some unique identifier which is used to retrieve the resource back from the data store via apis. So, ASA project id was not set as any attribute. But, we need "ASA Project Id" while creating resources like "AD Task Settings". We can pass name too but then it will require one extra "Get Project by name" call to get the project id.

**Concatenating multiple resource Attribute Pattern** -  In few cases, we are concatenating ASA Identifiers even though we already have required attributes in the resource schema. For Ex - For Project Group resource, pattern for Terraform resource id is <Project Name>:<Group Name>. Project Group resource schema also has attributes for Project and Group name. 

To keep Id attribute consistent, my proposal is to set it to ASA resource Id. We can always retrieve the resource back via API as long as required attributes are set in resource schema. For that purpose, I created [PR](https://github.com/okta/terraform-provider-oktapam/pull/69). I didn't change it currently for all the resources as I first want to collect feedback and see if we agree with the proposal.

Reply:
Removed project id attribute and populated ASA UUID for "id" attribute instead of name.